lib/src/test/java/io/ably/lib/test/rest/RestAuthAttributeTest.java

```diff
@@ -0,0 +1,109 @@
+     * <p>
+     * Spec: RSA10g
+     * </p>
+     */
+    @Test
+    public void auth_stores_options_params() {
+        try {
+            /* init custom TokenParams */
+            Capability capability1 = new Capability();
+            capability1.addResource("testchannel", "subscribe");
+            TokenParams tokenParams = new TokenParams() {{
+                ttl = 1000L;
+                clientId = "firstClientId";
+                capability = capability1.toString();
+            }};
+
+            /* init custom AuthOptions */
+            Setup.TestVars testVars = Setup.getTestVars();
+            AuthOptions authOptions = new AuthOptions() {{
+                authCallback = new TokenCallback() {
+                    private AblyRest ably = new AblyRest(testVars.createOptions(testVars.keys[0].keyStr));
+
+                    @Override
+                    public Object getTokenRequest(TokenParams params) throws AblyException {
+                        return ably.auth.requestToken(params, null);
+                    }
+                };
+                authUrl = "auth_url_1";
+                token = "test_token_1";
+                key = testVars.keys[1].keyStr;
+                tokenDetails = new TokenDetails();
+                authHeaders = new Param[]{new Param("X-Header-Param", "1")};
+                authParams = new Param[]{new Param("X-Auth-Param", "1")};
+            }};
+
+            /* authorise with custom options */
+            ably.auth.authorise(authOptions, tokenParams);
+
+            /* get stored TokenParams */
+            TokenParams storedTokenParams = ably.auth.getTokenParams();
+
+            /* Verify that,
+             * storedTokenParams isn't null,
+             * and storedTokenParams equals custom tokenParams */
+            assertNotNull(storedTokenParams);
+            assertEquals(storedTokenParams.ttl, tokenParams.ttl);
+            assertEquals(storedTokenParams.clientId, tokenParams.clientId);
+            assertEquals(storedTokenParams.capability, tokenParams.capability);
+
+            /* get stored AuthOptions */
+            AuthOptions storedAuthOptions = ably.auth.getAuthOptions();
+
+            /* Verify that,
+             * storedAuthOptions isn't null,
+             * and storedAuthOptions equals custom authOptions */
+            assertNotNull(storedAuthOptions);
+            assertEquals(storedAuthOptions.authCallback, authOptions.authCallback);
+            assertEquals(storedAuthOptions.authUrl, authOptions.authUrl);
+            assertEquals(storedAuthOptions.token, authOptions.token);
+            assertEquals(storedAuthOptions.key, authOptions.key);
+            assertEquals(storedAuthOptions.tokenDetails, authOptions.tokenDetails);
+            assertArrayEquals(storedAuthOptions.authHeaders, authOptions.authHeaders);
+            assertArrayEquals(storedAuthOptions.authParams, authOptions.authParams);
```

Comment:
Checking store auth options and stored token params is nice, but the spec states "Stores the AuthOptions and TokenParams arguments as defaults for subsequent authorisations".  You are not testing this as you have not made subsequent authorisations.  Please make a subsequent auth without any details to see that the details are actually used.